modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/tree/BplusTree.java

```diff
@@ -893,11 +897,12 @@ protected BplusTree(
             long metaPageId,
             @Nullable ReuseList reuseList,
             IoVersions<? extends BplusInnerIo<L>> innerIos,
-            IoVersions<? extends BplusLeafIo<L>> leafIos
+            IoVersions<? extends BplusLeafIo<L>> leafIos,
```

Comment:
I feel like passing a registry and types should be more convenient, what do you think?

Types can even be retrieved from protected methods to reduce a number of parameters in constructor. But if you think that it's a bad solution, just let me know. Invoking virtual methods in constructors is an anti-pattern, I know it

Reply:
Ok, that's fair. I guess we can revisit it in the future, it's not like the code can't be changed